theano/gof/vm.py

```diff
@@ -179,6 +186,13 @@ def __call__(self):
                     t1 = time.time()
                     self.call_counts[i] += 1
                     self.call_times[i] += t1 - t0
+                    for ins in node.inputs:
+                        info = self.reallocated_info.get(ins, None)
+                        if info:
+                            for outs in info[1]:
+                                if not outs[0]:
```

Comment:
Why this condition? If it is already allocated, why not still override it? I think we should overried it even if some space is already there. I'm not 100% sure, so what do you think? If we can do that, we won't need the loop, as we will always use the first element of info[1]